src/ipc/ipc-common.c

```diff
@@ -214,6 +214,9 @@ void ipc_msg_send(struct ipc_msg *msg, void *data, bool high_priority)
 	    msg->tx_data != data) {
 		ret = memcpy_s(msg->tx_data, msg->tx_size, data, msg->tx_size);
 		assert(!ret);
+		if (!cpu_is_primary(cpu_get_id()))
+			dcache_writeback_region((__sparse_force void __sparse_cache *)msg->tx_data,
+						msg->tx_size);
```

Comment:
This is a good catch. I wonder if a better fix would be to just allocate tx_data as uncached.

Reply:
Possible, but it probably comes with a performance cost. It's hard for me to determine what is better. However, invalidating each message also seems problematic to me, and I considered adding some flag in the structure that would indicate that the payload should be invalidated before copying.